### PR TITLE
Scratchpads: Dynamic position and size, depending on monitor

### PR DESCRIPTION
**Pull request by iliayar**
Hello. I have two monitors with different sizes and currently scratchpad's windows don't change it's size accordingly. So I've added an option to resize and move window depending on monitor size. In my config it looks like this:
```json
{
  "pyprland": {
    "plugins": [
      "scratchpads"
    ]
  },
  "scratchpads": {
    "term_quake": {
      "command": "wezterm start --class term_quake",
      "position": "0% 0%",
      "size": "100% 50%"
    }
  }
}
```
So on each monitor the position of terminal is at top left corner and it always a half of screen's height. Would you mind adding these options?

I've also add a nix flake, it will allow Nix users to install it directly.

### CODE REVIEW

File: pyprland/plugins/scratchpads.py [hunk 1]
```diff
@@ -394,22 +391,46 @@ async def run_show(self, uid, force=False) -> None:
...
+        """
+        Converts a string like "X Y" to coordinates relative to monitor
+        Supported formats for X, Y:
+        - Percentage: "V%". V in [0; 100]
+
+        Example:
+        "10% 20%", monitor 800x600 => 80, 120
+        """
+
+        if not coords:
```
Comment (fdev31) [inline pyprland/plugins/scratchpads.py:418]: Can we show an error log here ?

Comment (fdev31) [reply pyprland/plugins/scratchpads.py:418]: Or just an assert instead of those two lines...

**Review by fdev31 — CHANGES_REQUESTED** [review]
That's great!
Just one tiny change and I'll merge it, sorry for taking forever to see this PR.

Can you also suggest a short documentation which I'll add to the wiki, or an update to the existing doc to expose an example?

https://github.com/hyprland-community/pyprland/wiki/Plugins#scratchpads

File: pyprland/plugins/scratchpads.py (hunk 1 above)
Comment (iliayar) [reply pyprland/plugins/scratchpads.py]: I've reworked `_convert_coords` a bit. Now we check presence of `scale` and `position` in config before calling it.

**Comment by iliayar** [comment]
The description of new parameters:

#### `size` (optional)

string in format `"X% Y%"`, where X and Y is percentage of monitor's width and height accordingly. Every time scratchpad is shown, window will be resized depending on the monitor size, it displayed on.
For example on monitor of size `800x600` and `"size": "80% 80%"` in config scratchpad always have size `640x480`, regardless of which monitor it was first launched on.

#### `position` (optional)

every time scratchpad is shown, window will be moved to specified position relative to top left corner. For format and example see `size`.

example of scratchpad that always occupy top half of the screen:
```json
"scratchpads": {
    "term_quake": {
        "command":"wezterm start --class term_quake",
        "position":"0% 0%",
        "size":"100% 50%"
    }
}
```

**Comment by fdev31** [comment]
Nice, I'll look into it later this week.
One small question, `size` is currently achievable using windowrules in hyprland.conf isn't it ?
In the past I wondered if I should add support for such in pypr to make it more convenient to configure or not... I guess the current way can be improved if you felt the need for it.

**Comment by iliayar** [comment]
The problem with windowrules is that it's calculating once, when the window created. I didn't manage to find a way to force recalculation. For me personally it's more convenient to keep those two options in the same place, as this is more the property of scrathpad itself.

**Pull request merged**